garden_ai/model_connectors/model_utils.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, Type, Union, Optional
+
+from pydantic import (
+    HttpUrl,
+)
+
+from .exceptions import (
+    ConnectorInvalidUrlError,
+    UnsupportedConnectorError,
+)
+from .github_conn import GitHubConnector
+from .hugging_face import HFConnector
+from .model_connector import ModelConnector, ModelRepository
+
+
+# Maps ModelRepositorys to their corresponding connector class.
+CONNECTOR_MAPPING: Dict[ModelRepository, Type[ModelConnector]] = {
+    ModelRepository.GITHUB: GitHubConnector,
+    ModelRepository.HUGGING_FACE: HFConnector,
+}
+
+
+def match_connector_type_by_url(url: str) -> Optional[Type[ModelConnector]]:
```

Comment:
since we probably don't need them to be used outside of this module, I like the convention of prefixing helpers with an underscore to signify that they're internal